app/lib/actions/middleware/backend_services_check.rb

```diff
@@ -0,0 +1,32 @@
+module Actions
+  module Middleware
+    class BackendServicesCheck < Dynflow::Middleware
+      def plan(*args)
+        #To prevent the ping from happening multiple times, keep track on the initial entry action
+        parent = source_action
+        if Setting[:check_services_before_actions] && !parent.input[:backend_services_checked]
+          result = ::Katello::Ping.ping[:services]
+
+          ::Katello::Ping::SERVICES.each do |service|
+            if result[service][:status] != ::Katello::Ping::OK_RETURN_CODE
+              fail _("There was an issue with the backend service %s: ")  % service + result[service][:message]
+            end
+          end
+          parent.input[:backend_services_checked] = true
+        end
+
+        pass(*args)
+      end
+
+      private
+
+      def source_action
+        parent = self.action
+        until parent.trigger.nil?
+          parent = parent.trigger
+        end
+        parent
+      end
```

Comment:
You can use the `parent.root_action?` https://github.com/Dynflow/dynflow/blob/3c0dc068b7e77d1f25364d678f9e917753a564a4/lib/dynflow/action.rb#L517